spec/controller_services/shopping_list_items_controller/update_service_spec.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'service/ok_result'
+require 'service/not_found_result'
+require 'service/method_not_allowed_result'
+require 'service/unprocessable_entity_result'
+
+RSpec.describe ShoppingListItemsController::UpdateService do
+  describe '#perform' do
+    subject(:perform) { described_class.new(user, list_item.id, params).perform }
+
+    let(:user) { create(:user) }
+    let(:shopping_list) { user.shopping_lists.create! }
+    
+    context 'when all goes well' do
+      let!(:list_item) { create(:shopping_list_item, list: shopping_list, quantity: 2) }
+      let(:params) { { quantity: 3 } }
+      let(:master_list) { user.master_shopping_list }
+      let(:scope) { ShoppingListItem.belonging_to_user(user) }
+
+      before do
+        master_list.add_item_from_child_list(list_item)
+      end
+
+      it 'updates the shopping list item', :aggregate_failures do
+        perform
+        expect(list_item.reload.quantity).to eq 3
+        expect(list_item.reload.notes).to be nil
+      end
+
+      it 'updates the item on the master list' do
+        # Put the mocks in here because I don't want this much mocking in the other specs
+        allow(ShoppingListItem).to receive(:belonging_to_user).with(user).and_return(scope)
```

Comment:
These mocks ended up being necessary in a couple places due to the fact that I didn't want to test the whole `MasterListable` methods (since these are unit tests and they're covered in other unit tests). Unfortunately, because there are DB calls made in the controller service to fetch these objects, they aren't the same objects in memory as the ones in the spec. So I had to make sure the ones in the spec were getting passed around instead.